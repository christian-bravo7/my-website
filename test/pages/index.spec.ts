import IndexPage from '@/pages/index.vue';
import PortalVue from 'portal-vue';
import VueMeta from 'vue-meta';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(PortalVue);
localVue.use(VueMeta, { keyName: 'head' });

describe('index page', () => {
  it('should render page correctly', () => {
    const wrapper = mount(IndexPage, {
      localVue,
      mocks: {
        $t: jest.fn(msg => msg),
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display meta-tag title', () => {
    const wrapper = mount(IndexPage, {
      localVue,
      mocks: {
        $t: jest.fn(msg => msg),
      },
    });

    const title = wrapper.vm.$meta().refresh().metaInfo.title;

    expect(title).toBe('navigation.home-label');
  });
});
