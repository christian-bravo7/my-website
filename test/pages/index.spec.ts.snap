// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index page should render page correctly 1`] = `
<section>
  <div class="v-portal" style="display: none;"></div>
  <section class="section-container contact-me mb-12 md:mb-0">
    <div class="max-w-screen-lg | flex justify-center | my-0 mx-auto">
      <div class="grid grid-cols-1 md:grid-cols-2 gap-6">
        <div class="flex flex-col justify-center | mb-12 md:mb-0">
          <h3 class="mb-4 | section-title contact-me__title">
            Get in touch
          </h3>
          <p class="mb-14 mr-0 md:mr-8 md:mb-20 | font-base">
            Although I'm not currently looking for any new opportunities, my inbox is always open. Whether you have a question or just want to say Hello. I'll try my best to get back to you!
          </p>
          <div class="flex flex-col items-start"><span class="mb-2 | text-xs">
            Email me at:
          </span> <a href="mailto:christian.a.bravo.g@gmail.com" class="font-bold | font-base">
              christian.a.bravo.g@gmail.com
            </a></div>
        </div>
        <section class="flex justify-center items-center">
          <div class="grid gap-6 grid-cols-4 md:grid-cols-2 lg:grid-cols-4 page-social-icons"><a target="_blank" href="https://codepen.io/christian-bravo" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a><a target="_blank" href="https://github.com/christian-bravo7" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a><a target="_blank" href="https://www.instagram.com/christian.bravo_/" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a><a target="_blank" href="https://www.linkedin.com/in/christian151-bravo/" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a></div>
        </section>
      </div>
    </div>
  </section>
</section>
`;
